truss/model_framework.py

```diff
@@ -16,15 +17,20 @@ def typ(self) -> ModelFrameworkType:
         pass
 
     @abstractmethod
-    def infer_requirements(self):
-        """Mapping of requirements.txt line by name of the requirement.
+    def required_depedencies(self) -> Set[str]:
+        """Returns a set of packages required by this framework.
 
-        e.g. {'tensorflow': 'tensorflow==1.0.0'}
+        e.g. {'tensorflow'}
         """
         pass
 
-    def requirements_txt(self) -> List[str]:
-        return list(self.infer_requirements().values())
+    def requirements_txt(self, root_fn_name: str = "mk_truss") -> List[str]:
```

Comment:
Would be good to add a doc comment explaining the root_fn_name parameter. It's also worth thinking about whether root_fn_name is a good parameter.